### ISSUE TITLE
Supervised graph regression with Deep Graph CNN or GCN?

### ISSUE THREAD

**Issue by cappelchi** [stellargraph/stellargraph#1816]
How to perform supervised graph regression with Deep Graph CNN or GCN with single or multidimensional output?

**Comment by huonw** [comment]
Hi, we're keen to help. Could you say a bit more about your problem? Are you trying to take a collection of several graphs and predict attribute(s) for each one? Have you seen the demos for [graph classification](https://stellargraph.readthedocs.io/en/stable/demos/graph-classification/index.html)?

That's not _exactly_ what you're looking for, but fortunately classification and regression are very similar: typically one can change only the last output layer to turn a model from doing one of the tasks to the other. For example, the demos do binary classification and the last layer is:

```python
predictions = Dense(units=1, activation="sigmoid")(x_out)
```

A single-dimensional regression might change this to not use any activation, like `Dense(units=1)(x_out)`, or some other function that allows values outside 0 to 1. Similarly, for multi-dimensional output one can do the same things to replace the softmax activation normally used for multi-class classification.

How does that sound?

**Comment by cappelchi** [comment]
thanks.

**Comment by ltbd78** [comment]
So to summarize, do we only two changes to make it work?
1) No activation last layer
2) MSELoss

**Comment by huonw** [comment]
Yeah, that should work. Depending on the problem there may be other approaches, for instance using a `exponential` activation, if the output value is only positive, or using a different loss (e.g. MAE), but that's definitely where I'd start, personally.

**Comment by armp30** [comment]
I want it to learn the node scores can i use this library?
